### DISCUSSION TITLE
How to preserve selected index in a list?

### DISCUSSION THREAD

**Discussion by refaelsh** [jtdaugherty/brick#474] [Q&A] [answered]
Let's say I have a list that I am replacing its contents based on some external event.
How can I preserve the selected index?

Thanks in advance.

**Comment by jtdaugherty** [comment, marked as answer]
You can use [listReplace](https://hackage.haskell.org/package/brick-1.9/docs/Brick-Widgets-List.html#v:listReplace) to do that if get the selected index in advance and then pass it as the `Maybe Int` argument.

**Reply by refaelsh** [reply]
> get the selected index in advance

Yeap, that's what I was missing. Thank you!
